spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  let!(:user) { create(:user) }
+  let!(:post_valid_params) { attributes_for(:post)  }
+  let!(:post_invalid_params) { { description: 'a' * 2001, image: '' } }
+
```

Comment:
make two contexts: 'unauthorized user' and 'authorized user' which will be common for all actions